### PR TITLE
Auto save a new authenticated user

### PR DESCRIPTION
**Pull request by knockknockyoo**
BENTO-1897

1. Created an user model
2. Stored a user if not existed in the database
3. Added an utility to verify an element in the array
4. Added test codes

### CODE REVIEW

**Review by n2iw — CHANGES_REQUESTED** [review]
When saving a new user, system shouldn't try to  send emails to admin or the user. Please test with an empty Neo4j, and make sure after login and calling getMyUser, a new user recorded is created in Neo4j.

File: data-management/data-interface.js [hunk 1]
```diff
@@ -93,48 +100,47 @@ const listArms = async (input, context) => {
...
-                firstName: response.firstName,
-                lastName: response.lastName
-            }
-            await sendAdminNotification(adminEmails, template_params);
-            await sendRegistrationConfirmation(response.email, template_params)
-            return response;
-        }
-        else {
-            return new Error(errorName.UNABLE_TO_REGISTER_USER);
+        const notify = (parameters.isNotify === false) ? parameters.isNotify: true;
```
Comment (knockknockyoo) [inline data-management/data-interface.js:128]: fixed not to send an email notification after auto-generated user registration

**Review by knockknockyoo — COMMENTED** [review]
fixed not to send an email notification after auto-generated user registration

**Review by n2iw — CHANGES_REQUESTED** [review]
1. Please make new user have role of "non-member", and status of ""
2. Please don't set editDate for a new user.
3. Please rename user's "status" to "userStatus"

File: data-management/neo4j-service.js [hunk 2]
```diff
@@ -132,10 +133,11 @@ async function registerUser(parameters) {
             firstName: $firstName,
             lastName: $lastName,
             email: $email,
-            IDP: $IDP,
+            IDP: $idp,
             organization: $organization,
             userID: $userID,
-            registrationDate: $registrationDate,
+            creationDate: '${getTimeNow()}',
+            editDate: '',
```
Comment (knockknockyoo) [inline data-management/neo4j-service.js:140]: new user does not set edit date

File: graphql/schema.graphql [hunk 3]
```diff
@@ -11,7 +11,7 @@ type User {
     "admin, member (has ever been granted any accesses), non-member (never granted any access)"
     role: String
     "null (login only), inactive (all accesses revoked), active, disabled, deleted"
-    status: String
+    userStatus: String
```
Comment (knockknockyoo) [inline graphql/schema.graphql:14]: userstatus field changed

File: data-management/neo4j-service.js [hunk 4]
```diff
@@ -132,11 +133,12 @@ async function registerUser(parameters) {
...
             email: $email,
-            IDP: $IDP,
+            IDP: $idp,
             organization: $organization,
             userID: $userID,
-            registrationDate: $registrationDate,
-            status: $status,
+            creationDate: '${getTimeNow()}',
+            editDate: '',
+            userStatus: $status,
```
Comment (knockknockyoo) [inline data-management/neo4j-service.js:141]: userstatus field changed

**Review by knockknockyoo — COMMENTED** [review]
1. for member "non-member", and status of ""
2. editDate for a new user.
3. update "status" to "userStatus" field and graphql schema

File: data-management/data-interface.js [hunk 5]
```diff
@@ -93,48 +100,47 @@ const listArms = async (input, context) => {
...
+    const task = async () => {
+        inspectValidUser(parameters);
+        if (!await checkUnique(parameters.userInfo.email, parameters.userInfo.idp)) throw new Error(errorName.NOT_UNIQUE);
+
         let generatedInfo = {
             userID: v4(),
-            registrationDate: (new Date()).toString(),
-            status: "registered",
-            role: "standard"
+            status: NONE,
```
Comment (knockknockyoo) [inline data-management/data-interface.js:120]: a new register user assigned as NONE of user-status as empty string ""

**Review by knockknockyoo — COMMENTED** [review]
fixed the changes
1. user-role
2. user-status

File: constants/user-constant.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+module.exports = Object.freeze({
+    // user status
+    NONE: '',
+    INACTIVE: 'inactive',
```
Comment (knockknockyoo) [inline constants/user-constant.js:4]: added inactive

**Review by knockknockyoo — COMMENTED** [review]
added inactive & active

**Review by n2iw — APPROVED** [review]

**Pull request merged**